src/hardware/opl.cpp

```diff
@@ -578,6 +579,7 @@ uint8_t Opl::AdlibGoldControlRead()
 
 void Opl::PortWrite(const io_port_t port, const io_val_t value, const io_width_t)
 {
+	std::lock_guard lock(mutex);
 	RenderUpToNow();
```

Comment:
In PS/1 audio, the lock is inside the `Render(..)` call, but in some like here in OPL and Tandy, it's in the port write prior to Render().

Is the difference that we need to protect any subsequent device register writes, so those are atomic as well?

Reply:
That was an oversight in PS/1.  We should be locking around the port write to prevent the register state from being change mid-render.  Moved PS/1 audio lock to port write.